TW-235: remove wildcards from Visibility docs (#1919)
temporalio/documentation: docs/visibility.md

- **=, !=, >, >=, <, <=**
- **AND, OR, ()**
- **BETWEEN ... AND**
- **IN**
- **ORDER BY**

The **ORDER BY** operator is supported only when Elasticsearch is used as the Visibility store.

### Partial string match

The `=` operator works like **CONTAINS** to find Workflows with Search Attributes that contain a specific word.

<!-- note: advanced vis features will be supported in SQL upon the release of v1.20.-->

For example, if you have a Search Attribute `Description` of `Text` type with the value of "The quick brown fox jumps over the lazy dog", searching for `Description=quick` or `Description=fox` will successfully return the Workflow.
However, partial word searches such as `Description=qui` or `Description=laz` will not return the Workflow.
This is because [Elasticsearch's tokenizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-standard-tokenizer.html) is configured to return complete words as tokens.

### Efficient API usage

An Advanced List Filter API may take longer to respond if it is retrieving a large number of Workflow Executions (over 10,000).

> <!-- Use the `CountWorkflow` API to efficiently count the number of <a class="tdlp" href="/workflows#workflow-execution">Workflow Executions<span class="tdlpiw"><img src="/img/link-preview-icon.svg" alt="Link preview icon" /></span><span class="tdlpc"><span class="tdlppt">What is a Workflow Execution?</span><br /><br /><span class="tdlppd">A Temporal Workflow Execution is a durable, scalable, reliable, and reactive function execution. It is the main unit of execution of a Temporal Application.</span><span class="tdlplm"><br /><br /><a class="tdlplma" href="/workflows#workflow-execution">Learn more</a></span></span></a>. -->

Reviewer: ditto

Author: Generated document. No need to comment on this one since the changes are made in the source file above.

  - **BETWEEN ... AND**
  - **IN**

>   - **ORDER BY**

Reviewer 2: Server v1.20 will support Advanced Visibility capabilities on SQL databases; Will be updating this in a diff PR though; this is just FYI.

- ORDER BY is only supported on Elasticsearch and not on the SQL dbs. 
- Same with LIKE. It only works with Elasticsearch, and on Search attributes of type text IIRC. I think LIKE is still supported but is a major point of confusion with regards to usage. We can either remove it completely (which I see you have) or add clarifications etc. and only remove after support for it is dropped in a later release.

Author: I've already removed it, so I'll keep it as is. According to Alex, it doesn't work as one would hope.

>   The last operator (**ORDER BY**) is only supported with Elasticsearch as a Visibility store.

Reviewer 2: ```suggestion
  The last operator (**ORDER BY**) is only supported with Elasticsearch integration.
```
